scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,579 @@
+            System.out.print("groups for user " + userInfo.getLogin() + ":");
+            String groupString = console.readLine();
+            userInfo.setGroups(groupString.split(","));
+        }
+    }
+
+    /**
+     * Update the accounts in the login and config files
+     *
+     * @throws ManageUsersException
+     */
+    private static void updateAccounts(PublicKey pubKey, UserInfo userInfo, String loginFilePath, String groupFilePath, Action action) throws ManageUsersException {
+        try {
+            Properties props = new Properties();
+            try {
+                FileInputStream stream = new FileInputStream(new File(loginFilePath));
+                props.load(stream);
+                stream.close();
+            } catch (Exception e) {
+                exitWithErrorMessage("could not read login file : " + loginFilePath, null, e);
+            }
+
+            Multimap<String, String> groupsMap = loadGroups(groupFilePath);
+
+            switch (action) {
+                case CREATE:
+                    if (props.containsKey(userInfo.getLogin())) {
+                        exitWithErrorMessage("USER already exists in login file : " + loginFilePath, null, null);
+                    }
+                    updateUserPassword(pubKey, userInfo.getLogin(), userInfo.getPassword(), props);
+                    System.out.println("Created user " + userInfo.getLogin() + " in " + loginFilePath);
+                    updateUserGroups(userInfo.getLogin(), userInfo.getGroups(), groupsMap);
+                    break;
+                case UPDATE:
+                    if (!props.containsKey(userInfo.getLogin())) {
+                        exitWithErrorMessage("USER does not exist in login file : " + loginFilePath, null, null);
+                    }
+                    if (userInfo.getPassword() != null) {
+                        updateUserPassword(pubKey, userInfo.getLogin(), userInfo.getPassword(), props);
+                        System.out.println("Changed password for user " + userInfo.getLogin() + " in " + loginFilePath);
+                    }
+                    if (userInfo.getGroups() != null) {
+                        updateUserGroups(userInfo.getLogin(), userInfo.getGroups(), groupsMap);
+                    }
+                    break;
+                case DELETE:
+                    if (!props.containsKey(userInfo.getLogin())) {
+                        exitWithErrorMessage("USER does not exist in login file : " + loginFilePath, null, null);
+                    }
+                    props.remove(userInfo.getLogin());
+                    groupsMap.removeAll(userInfo.getLogin());
+                    System.out.println("Deleted user " + userInfo.getLogin() + " in " + loginFilePath);
+                    break;
+            }
+
+            storeLoginFile(loginFilePath, props);
+
+            storeGroups(groupFilePath, groupsMap);
+        } catch (Throwable t) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1181)